src/GafferScene/ShaderPlug.cpp

```diff
@@ -139,12 +119,43 @@ bool ShaderPlug::acceptsInput( const Gaffer::Plug *input ) const
 
 IECore::MurmurHash ShaderPlug::attributesHash() const
 {
-	const Shader *s = shader( this );
-	return s ? s->attributesHash() : MurmurHash();
+	const Gaffer::Plug *p = shaderOutPlug();
+
+	IECore::MurmurHash h;
+	if (p)
+	{
+		const GafferScene::Shader *s = runTimeCast<const GafferScene::Shader>( p->node() );
```

Comment:
Seems like it'd make sense to put this check in `shaderOutPlug()` rather than repeat it in both callers? The function name implies it would only return ever return a shader plug...